data/json/mapgen/orchard_apple.json

```diff
@@ -328,18 +273,103 @@
         { "monster": "GROUP_DOMESTIC", "x": [ 1, 22 ], "y": [ 1, 22 ], "chance": 50 },
         { "monster": "GROUP_SAFE", "x": [ 1, 22 ], "y": [ 1, 22 ], "chance": 20 },
         { "monster": "GROUP_ZOMBIE", "x": [ 1, 22 ], "y": [ 1, 22 ], "chance": 25 }
       ]
     }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ "orchard_processing_roof" ],
+    "object": {
+      "fill_ter": "t_tile_flat_roof",
+      "rows": [
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "   |22222222222222223   ",
+        "   |................3   ",
+        "   |.:...........7..3   ",
+        "   |................3   ",
+        "   |..6......###....3   ",
+        "   |.........#z#....3   ",
+        "   |.........###....3   ",
+        "   |................3   ",
+        "   |................3   ",
+        "   |................3   ",
+        "   |................3   ",
+        "   |---5------------3   ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        "
+      ],
+      "palettes": [ "roof_palette" ],
+      "terrain": {
+        "z": "t_ladder_down"
+      },
+      "nested": {
+        "6": { 
+          "chunks": [ [ "null", 20 ], [ "roof_6x6_garden_3", 80 ] ]
+        },
+        "7": {
+          "chunks": [ [ "null", 20 ], [ "roof_2x2_infrastructure", 40 ], [ "roof_2x2_infrastructure_2", 40 ] ]
+        }
+      }
+    }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ "orchard_stall_roof" ],
+    "weight": 500,
+    "object": {
+      "fill_ter": "t_tile_flat_roof",
+      "rows": [
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "            |5222222223 ",
+        "            |.........3 ",
+        "            |.6.....7.3 ",
+        "            |.........3 ",
+        "            |.......7.3 ",
+        "            |.........3 ",
+        "            |.........3 ",
+        "            |.........3 ",
+        "            ----------3 ",
+        "                        ",
+        "                        "
+      ],
+      "palettes": [ "roof_palette" ],
+      "nested": {
+        "6": {
+          "chunks": [ [ "null", 30 ], [ "roof_4x4_holdout", 40 ], [ "roof_5x5_coop", 40 ] ]
```

Comment:
For nested chunks with larger items like the coop and holdout, make sure there's logical roof access via a ladder or stairs.